build/azure-pipelines/ci-build-test.yml

```
@@ -37,6 +39,57 @@ stages:
         packageType: 'sdk'
         version: '9.x'
 
+    # Check what type of changes were made to optimize pipeline execution
+    - bash: |
+        echo "Analyzing changes to optimize pipeline execution..."
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)